### PR TITLE
send fee per gas to checkUpkeep

### PR DESCRIPTION
**Pull request by MStreet3**
send appropriate fees to `eth_call` tasks.
prevent nodes from being keepers on chains that support eip1559 if node is not configured to support eip1559 fees.

### todo
- [ ] remove schema references to upkeep feature flag
- [ ] move eip1559 support check to appropriate place

### CODE REVIEW

File: core/services/keeper/delegate.go [hunk 1]
```diff
@@ -111,3 +119,13 @@ func (d *Delegate) ServicesForSpec(spec job.Job) (services []job.Service, err er
...
+
+/* todo: move before merged to an appropriate location */
+func hasEIP1559Support(chainId string) bool {
+	switch chainId {
+	case "1":
+		return true
+	default:
+		return false
+	}
+}
```
Comment (begmaroman) [inline core/services/keeper/delegate.go:131]:
```
func hasEIP1559Support(chainId string) bool {
	return chainId == "1"
}
```

File: core/services/keeper/delegate.go [hunk 2]
```diff
@@ -111,3 +119,13 @@ func (d *Delegate) ServicesForSpec(spec job.Job) (services []job.Service, err er
 		upkeepExecuter,
 	}, nil
 }
+
+/* todo: move before merged to an appropriate location */
```
Comment (begmaroman) [inline core/services/keeper/delegate.go:123]:
```
// TODO: move before merged to an appropriate location
```

File: core/services/keeper/delegate.go [hunk 3]
```diff
@@ -95,6 +95,14 @@ func (d *Delegate) ServicesForSpec(spec job.Job) (services []job.Service, err er
...
 		SyncUpkeepQueueSize:      chain.Config().KeeperRegistrySyncUpkeepQueueSize(),
 	})
+
+	/* check that node has proper configuration for the given chain ID */
+	chainId := chain.Client().ChainID().Text(10)
+	if hasEIP1559Support(chainId) && !chain.Config().EvmEIP1559DynamicFees() {
+		return nil, errors.Errorf("cannot perform upkeeps on chain %d as it supports EIP1559 and node is not configured for EIP1559", chainId)
+	}
+
+	/* construct executer */
```
Comment (begmaroman) [inline core/services/keeper/delegate.go:105]:
```
// Construct executer
```

File: core/services/keeper/delegate.go [hunk 4]
```diff
@@ -95,6 +95,14 @@ func (d *Delegate) ServicesForSpec(spec job.Job) (services []job.Service, err er
 		Logger:                   svcLogger,
 		SyncUpkeepQueueSize:      chain.Config().KeeperRegistrySyncUpkeepQueueSize(),
 	})
+
+	/* check that node has proper configuration for the given chain ID */
```
Comment (begmaroman) [inline core/services/keeper/delegate.go:99]:
```
// Check that node has proper configuration for the given chain ID
```

File: core/services/keeper/delegate.go [hunk 5]
```diff
@@ -95,6 +95,14 @@ func (d *Delegate) ServicesForSpec(spec job.Job) (services []job.Service, err er
...
+	chainId := chain.Client().ChainID().Text(10)
+	if hasEIP1559Support(chainId) && !chain.Config().EvmEIP1559DynamicFees() {
+		return nil, errors.Errorf("cannot perform upkeeps on chain %d as it supports EIP1559 and node is not configured for EIP1559", chainId)
```
Comment (begmaroman) [inline core/services/keeper/delegate.go:102]:
```
return nil, errors.Errorf("cannot perform upkeeps on chain %s as it supports EIP1559 and node is not configured for EIP1559", chainId)
```

File: core/services/keeper/upkeep_executer.go [hunk 6]
```diff
@@ -184,14 +184,12 @@ func (ex *UpkeepExecuter) execute(upkeep UpkeepRegistration, headNumber int64, d
...
-			svcLogger.Error(errors.Wrap(err, "estimating gas price"))
-			return
-		}
-		gasPrice, gasTipCap, gasFeeCap = price, fee.TipCap, fee.FeeCap
+	price, fee, err := ex.estimateGasPrice(upkeep)
+	if err != nil {
+		svcLogger.Error(errors.Wrap(err, "estimating gas price"))
+		return
 	}
+	gasPrice, gasTipCap, gasFeeCap = price, fee.TipCap, fee.FeeCap
```
Comment (begmaroman) [inline core/services/keeper/upkeep_executer.go:192]: You can just use `price, fee.TipCap, fee.FeeCap` in the code below

File: core/services/keeper/delegate.go (hunk 3 above)
Comment (samsondav) [reply core/services/keeper/delegate.go:105]: Comment seems pointless?

File: core/services/keeper/delegate.go [hunk 7]
```diff
@@ -95,6 +95,14 @@ func (d *Delegate) ServicesForSpec(spec job.Job) (services []job.Service, err er
 		Logger:                   svcLogger,
 		SyncUpkeepQueueSize:      chain.Config().KeeperRegistrySyncUpkeepQueueSize(),
 	})
+
+	/* check that node has proper configuration for the given chain ID */
+	chainId := chain.Client().ChainID().Text(10)
+	if hasEIP1559Support(chainId) && !chain.Config().EvmEIP1559DynamicFees() {
```
Comment (samsondav) [inline core/services/keeper/delegate.go:101]: IMO don't bother checking it. Let the simulation fail instead and add a useful error message to the simulation fail that explains it.

In addition you might consider adding a check to the validation stage of when you add a job.

**Comment by cl-homu** [comment]
:umbrella: The latest upstream changes (presumably 78039a2897d4b4dda27d98aac59821f41eefd274) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

**Pull request closed without merging**